discretize/tree_mesh.py

```diff
@@ -359,6 +360,316 @@ def origin(self, value):
         # then update the TreeMesh with the hidden value
         self._set_origin(self._origin)
 
+    def refine_bounding_box(self, points, level, padding_cells_by_level=None, finalize=True, diagonal_balance=None):
+        """Refine within a bounding box based on the maximum and minimum extent of scattered points.
+
+        This function refines the tree mesh based on the bounding box defined by the
+        maximum and minimum extent of the scattered input `points`. It will refine
+        the tree mesh for each level given.
+
+        It also optionally pads the bounding box at each level based on the number of
+        padding cells at each dimension.
+
+        Parameters
+        ----------
+        points : (N, dim) array_like
+            The bounding box will be the maximum and minimum extent of these points
+        level : (n_level) array_like of int
+            The level of the treemesh to refine the bounding box to. If an array it will
+            refine the box to each level (mostly useful in combination with `padding_cells_by_level`)
+            Negative values index backwards, (e.g. `-1` is `max_level`).
```

Comment:
This is a bit annoying to have to count backward - will result in a lot of explanations. Can we not do the translation for the user?

Reply:
You don't have to count backwards, you could also supply the refinement level as well. The negative indexing works just like array indexing in python.

Reply:
I hear you, but still not very convenient.